### PR TITLE
Source playback

### PR DESCRIPTION
**Pull request by mjh1**
**What does this pull request do? Explain your changes. (required)**

**Specific updates (required)**

**How did you test each of these updates (required)**

**Does this pull request close any open issues?**

**Screenshots (optional)**

**Checklist**

- [ ] I have read the **CONTRIBUTING** document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

**Comment by mjh1** [comment]
@victorges aside from lack of test coverage, LGTU?

File: packages/api/src/controllers/asset.ts [hunk 1]
```diff
@@ -361,6 +361,12 @@ export async function withPlaybackUrls(
     console.error("Error getting asset object store", err);
     return asset;
   }
+  if (asset.sourcePlaybackReady) {
```
Comment (victorges) [inline packages/api/src/controllers/asset.ts]: An alternative could be also to just use the existence of `files` in the asset, specifically for a playback manifest. Those are only set on termination today and are now being set in the middle of processing. Not required change just wondering, WDYT?

File: packages/api/src/task/scheduler.ts [hunk 2]
```diff
@@ -207,6 +213,31 @@ export class TaskScheduler {
...
+    const asset = await db.asset.get(task.outputAssetId);
+    await this.updateAsset(task.outputAssetId, {
+      sourcePlaybackReady: true,
+      files: event.output.upload.assetSpec.files,
+      status: {
+        phase: asset.status?.phase,
+        progress: asset.status?.progress,
+        errorMessage: asset.status?.errorMessage,
+        updatedAt: Date.now(),
+      },
```
Comment (victorges) [inline packages/api/src/task/scheduler.ts]: **nit** What you want is probably
```suggestion
      status: {
        ...asset.status,
        updatedAt: Date.now(),
      },
```

File: packages/www/components/AssetDetails/AssetPlayerBox/index.tsx [hunk 3]
```diff
@@ -8,7 +8,7 @@ import FileUploadingProgress from "./FileUploadingProgress";
 import ProcessingProgress from "./ProcessingProgress";
 
 const AssetStatusBox = ({ asset }: { asset?: Asset }) => {
-  if (asset?.status?.phase === "ready" && asset.playbackUrl) {
+  if (asset?.playbackUrl) {
```
Comment (victorges) [inline packages/www/components/AssetDetails/AssetPlayerBox/index.tsx:11]: Hah nice, makes the most sense

**Review by victorges — APPROVED** [review]
Yeah LGTM!

File: packages/api/src/controllers/asset.ts (hunk 1 above)
Comment (mjh1) [reply packages/api/src/controllers/asset.ts]: Yeah I was thinking the same really. Does this work: https://github.com/livepeer/studio/pull/1748/commits/a88980c404b69e9ace01d3ecfa8a7fad638e6d5d
or maybe i should also check the return from `getPlaybackUrl` is not undefined

Comment (mjh1) [reply packages/api/src/controllers/asset.ts]: Actually no if it's undefined then it'll be hidden from the response object anyway

**Comment by mjh1** [comment]
@victorges Does the test look ok to you?  https://github.com/livepeer/studio/pull/1748/commits/3099be80033adf723a2c2c8988c3a2db96d49ace

**Comment by victorges** [comment]
@mjh1 it does! Thanks for creating that, will make it much easier to create future tests on the scheduler:)

File: packages/api/src/controllers/asset.ts (hunk 1 above)
Comment (victorges) [reply packages/api/src/controllers/asset.ts]: Yeah like that exactly! Does it work fine? Your tests still seemed to check the existence of the boolean

Comment (mjh1) [reply packages/api/src/controllers/asset.ts]: Yep worked ok, but yes i'm still using the bool in one other place so will do a follow up PR

**Pull request merged**